openwisp_controller/connection/admin.py

```diff
@@ -166,6 +167,7 @@ def schema_view(self, request):
 
 
 DeviceAdmin.inlines += [DeviceConnectionInline]
+reversion.register(model=DeviceConnection, follow=['device'])
```

Comment:
I guess we may have to do something similar also in openwisp-monitoring and openwisp-firmware-upgrade? If so can you please open issues there too (to be done after this)?

Reply:
I insist with this, please let me know if you have created the issues already.

Reply:
In openwisp-monitoring, device status, device chart, checks and alert settings are not recovered while restoring a device. But I observed no error in the recover operation. 

Similarly in openiwsp-firmware-upgrader, firmware image and recent firmware upgrade models were missing after recovering a deleted device. But again, no errors were encountered in the recover operation.